### PR TITLE
Fix pyright errors in /tasks

### LINKED ISSUE #576: Resolve Pyright errors for Tasks

### PR DESCRIPTION
**Pull request by collindutter**
Resolves #576

### CODE REVIEW

**Review by cjkindel — DISMISSED** [review]

File: griptape/tasks/action_subtask.py [hunk 1]
```diff
@@ -47,17 +46,33 @@ class ActionSubtask(PromptTask):
     def input(self) -> TextArtifact:
         return TextArtifact(self._input)
 
+    @input.setter
+    def input(self, value: str | TextArtifact | Callable[[BaseTask], TextArtifact]) -> None:
+        if isinstance(value, str):
+            self._input = value
+        else:
+            raise Exception("ActionSubtask input must be a string.")
```
Comment (andrewfrench) [inline griptape/tasks/action_subtask.py:54]: I think the signature and the implementation are misleading here

File: griptape/tasks/prompt_task.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ class PromptTask(BaseTextInputTask):
         default=Factory(lambda self: self.default_system_template_generator, takes_self=True), kw_only=True
     )
 
-    output: TextArtifact | ErrorArtifact | Optional[InfoArtifact] = field(default=None, init=False)
+    output: BaseArtifact = field(default=None, init=False)
```
Comment (andrewfrench) [inline griptape/tasks/prompt_task.py:20]: Optional?

**Review by andrewfrench — CHANGES_REQUESTED** [review]
comments above

File: griptape/tasks/action_subtask.py (hunk 1 above)
Comment (collindutter) [reply griptape/tasks/action_subtask.py:54]: Maybe we should just remove this override? It's not really clear to me why it was overridden in the first place.

File: griptape/tasks/prompt_task.py (hunk 2 above)
Comment (collindutter) [reply griptape/tasks/prompt_task.py]: Good catch

File: griptape/tasks/action_subtask.py (hunk 1 above)
Comment (andrewfrench) [reply griptape/tasks/action_subtask.py:54]: Looks like it'd be fine if we removed the override.

Comment (collindutter) [reply griptape/tasks/action_subtask.py]: Updated. The key part of the override is that we don't render a jinja template. We want the input exactly as it is.

**Review by andrewfrench — APPROVED** [review]

**Pull request merged**